### ISSUE TITLE
Specific id "pub01" prevents list item from being shown [FRA: AdGuard Français]

### ISSUE THREAD

**Issue by malparti** [AdguardTeam/AdguardFilters#86103]
It appears that the filter FRA: AdGuard Français blocks any item with id `pub01`. See the discussion there: https://github.com/uBlockOrigin/uAssets/issues/9429

I think this drastic filter is very risky because "pub" can be used as a three-letter abbreviation for the fairly generic word *publication* (which could refer to blog entries, books, scientific articles, etc). Thus, the potential for having a legitimate item called `pub01` is fairly high -- in fact probably higher than having it refer to an ad if we are talking about a website in English (or pretty much any language other than French).

It is also slightly surprising that this seems to occur with `pub01` but not, e.g, with `pub00` and/or `pub02`.

**Comment by krystian3w** [comment]
@smed79

**Comment by Alex-302** [comment]
https://github.com/uBlockOrigin/uAssets/issues/9429#issuecomment-862567040

**Comment by krystian3w** [comment]
I you found too many pages maybe good exclude rule on network level:

https://github.com/AdguardTeam/FiltersRegistry/blob/master/filters/exclusions.txt

as "too wide rule.

**Comment by malparti** [comment]
@Alex-302 I do not understand why you closed the issue. The comment that you link say it has been fixed, but the commit preceding that comment, namely 6539bd8, does not fix it. As I mentioned, the domain you white-listed isn't even a "real" domain containing webpages that anyone is going to consult.

**Comment by krystian3w** [comment]
uBo support drop domains/tld before `#@#` if edit `exclusions.txt` need big debate.

**Comment by Alex-302** [comment]
Nothing is broken now
![image](https://user-images.githubusercontent.com/8361299/122287612-cd365280-cef9-11eb-988a-f59052342fc8.png)
No need to remove the rule from filter.


> the domain you white-listed isn't even a "real"

Report the real site, not testing pages.

**Comment by Alex-302** [comment]
> I you found too many pages

Examples?

**Comment by malparti** [comment]
@Alex-302

> Report the real site, not testing pages.

The issue I am trying to report is not specific to a given site, it is a general problem associated to the fact of systematically blocking html elements with id `pub01`, which seems to be what uBO started doing recently. It wasn't doing that before.

I have explained why this could be a problem. It did cause some issues on webpages I maintain, which is how I noticed the issue. I was able to notice the issue only because I knew precisely how many elements were supposed to be in the list. But if it happens on a page whose content you do not know, you have no reason to notice -- there will only be one publication missing. This is what makes that especially problematic.

I cannot give you an example because I changed the html ids on every page I maintain to make sure the issue would be fixed quickly.

White-listing is not a good solution because there is no way to list all of the pages that could make valid use of the id "pub01". "pub" is the three letter abbreviation of one an incredibly common word, namely *publication*.

I believe this iissue is fairly recent, so most likely something in FRA: AdGuard Français that was not a good idea. That is the issue I am trying to report.

**Comment by Alex-302** [comment]
Liste-FR is upstream of AdGuard French filter. @smed79 is author. https://github.com/easylist/listefr

> which seems to be what uBO started doing recently

That rule is 8years old https://github.com/easylist/listefr/commit/cb670d9310e5b16d75c08e4b71c1c38edd03d96e

**Comment by malparti** [comment]
@Alex-302

> That rule is 8 years old

On the webpage I maintain the problem started occurring at most a few months ago. It wasn't occurring on April 12th or at any time before that. I first noticed it a few days ago.

In all likeliness this is causing issues on a lot of webpages and goes under-reported. For instance, consider the following page, which I had never seen until 5 minutes ago:

http://aim-vil.com/category/recommend/concert_recommend/

Here two "buttons" are missing below the picture. Yet as a uBO user I would have never known there was something wrong without looking at the code.

**Comment by malparti** [comment]
The facts that
1. The rule is 8 years old yet the problem started occurring on my PC only at most two months ago.
2. On the bmoattachments.org example that I linked, `pub01` was blocked but not `pub02` whereas on the aim-vil.com link both `pub01` and `pub02` are blocked

Might indicated that there is a deeper problem.

**Comment by krystian3w** [comment]
> might actually work fine

nope, if "楽譜・出版物" means: 
*"Sheet music - publications" 
*  "Partitions (Remarques) et publications" 
*  "Ноты и публикации".

To disable range maybe possible use:

`##[lang*="fr"] #pub01`
`###pub01:lang(fr)`

worse if page no mark language in html code (very old pages or bad write new pages).

**Comment by malparti** [comment]
Regarding why the behavior changed recently even though the rule has been there for 8 years:
1. Is it possible that FRA was not activated by default before some version of Firefox/uBO? I never requested that filter nor looked at what filters were activated before running into that issue. I have recently upgraded Firefox to 78.11.0esr on Debian and that is when I first noticed the issue. As mentioned above, I am certain it was not occurring with my previous version of Firefox (which I think was 78.10, but might have been 78.9 or 78.8).
2. If FRA has always been activated by default on Firefox with French locale, could the change of behavior be explained by what language is inferred for webpage? The webpage on which I noticed the issue on has no declared language.

@krystian3w thanks for your help. 0n http://aim-vil.com/ it looks like the item named `pub01` is empty; this is the source code I see for the page:
```html
                                <div class="tab-pane fade" id="pub01" >
                                <!-- 物販 -->
                                                                <div class="row">
                                                                </div>
                                
                                							</div>
```
and the link to `#pub01` is displayed:
![2021-06-17_01-49](https://user-images.githubusercontent.com/85755504/122304840-8d2d9a80-cf0e-11eb-8739-359e2d5ec309.png)

So that website might not be a good example -- although it does illustrate the fact that some webdevelopper might legitimately want to call something `pub01`.

**Comment by Alex-302** [comment]
If something is broken - we add exclusions. If broken one unknown site, this is not a reason to remove the rule.

Comment
> ```
> 
> ```
is **`Product sales`**, I guess ad related.

**Comment by malparti** [comment]
@Alex-302 the point I've been trying to make is that this is probably broken on many websites and that it most likely goes under-reported. As an example of that, it took me about 5 minutes to find a website were there could potentially be a problem. Obviously, I'm not going to be able to give you a full list of every website on the web that uses `pub` as a short id for `publication`.

With all due respect, I think you are not trying to understand the problem. Let me rephrase it one last time in different terms: the FRA filter assumes that
1. `pub01`, when standing for "publicité 01", is frequent enough to require a special rule.
2. `pub02`, when standing for "publicité 02", is rare enough that it does not require a rule.
3. `pub01`, when standing for "publication 01", is rare enough not to prevent a rule.

I do not know what these assumptions are based on. I can easily believe (1), but I have a hard time believing (1) + (2) -- why would someone label something 01 if they're no going to have a 02 at some point -- and (3).

These assumptions broke a webpage I maintain, I suspect they can break many other. I've been trying to forward this to whoever is in charge of the uBO filters, but my concerns so far have been met with what -- to me -- kinda looks like denial / unwillingness to look into whether there really is an issue there.

Regarding your comment: I do not think it is ad related, I think it might just be something where they list their sheet-music and other publications.

Also -- and although this is an unrelated question -- none of this explains why the 8 year-old rule `pub01` suddenly started being active on my setting without me ever asking for it.

**Comment by malparti** [comment]
@Alex-302 Looking at the time between my two messages, you can see that it took me less that 10 minutes to find another website that is broken because of this filter:

http://www.simamills.in/publications/

The image associated to the first item of the list is blocked.

I've explained why this could cause a problem and I have given concrete examples to back this up. However, I'm not going to try to go and find every website uses `pub01` somewhere in their source code.

**Comment by Alex-302** [comment]
Literally every rule without domain restriction can break the sites. Just look to whitelist files.

> http://www.simamills.in/publications/

That rule cannot break this site. There are no elements with `ID`=`pub01`

**Comment by malparti** [comment]
> Literally every rule without domain restriction can break the sites. Just look to whitelist files.

Yes, but a rule called `item01` is going to break pretty much half of the sites and therefore is not a good rule. I think `pub01` is not a good rule, and I have provided arguments and evidence for that now.

> That rule cannot break this site.

Yet the site is broken by uBO, and the problem happens to be with a picture called `pub01`.

Without uBO:
![without_uBO](https://user-images.githubusercontent.com/85755504/122589651-34bee000-d069-11eb-9330-972cb92c6297.png)

With uBO:
![with_uBO](https://user-images.githubusercontent.com/85755504/122589620-2a9ce180-d069-11eb-90d3-16e4d0da732b.png)

**Comment by malparti** [comment]
I just saw that @smed79 added bmoattachments to the whitelist of listefr, but as I mentioned here and [there](https://git.io/JnEgN) this is not useful as that webpage is a test page I created to illustrate the issue.

No one has replied to the fact that `pub01` breaks several other websites, such as http://www.simamills.in/publications/ (although @Alex-302 seemed to disagree with that -- but see my screenshots). Do I need to find yet another broken page to convince people that the points I developed above make sense?

Conversely, are there really that many instances of `pub01` being used to refer to ads (rather than to a "publication" or something "public")?
